config.go

```diff
@@ -0,0 +1,83 @@
+package retry
+
+import (
+	"math/rand"
+	"time"
+)
+
+// Config represents the possible options when creating a Policy.  This type is friendly
+// to being unmarshaled from external sources.
+//
+// Three basic kinds of retry policies are created by this type:
+//
+//   - if Interval is nonpositive, the created policy will never retry anything
+//   - if Interval is positive but Jitter and Multiplier are not, the created policy will return a constant, unchanging retry interval
+//   - if Interval is positive and Jitter or Multiplier are as well, the created policy will return an exponentially increasing retry interval
+type Config struct {
+	// Interval specifies the retry interval for a constant backoff and the
+	// initial, starting interval for an exponential backoff.
+	//
+	// If this field is unset, no retries will happen.
+	Interval time.Duration `json:"interval" yaml:"interval"`
+
+	// Jitter is the random jitter for an exponential backoff.
+	//
+	// If this value is nonpositive, it is ignored.  If both this field and
+	// Multiplier are unset, the resulting policy will be a constant backoff.
+	Jitter float64 `json:"jitter" yaml:"jitter"`
```

Comment:
I understand the concepts behind Jitter, but I'm a bit confused about how it works in practice as a `float64`

Reply:
It's a percentage, plus/minus.  A jitter of 0.1 means choose a random value between 0.9x and 1.1x the base interval, which for an exponential backoff gradually increases according to some base multiplier.  You want a percentage, not an absolute value, because when backoffs happen you want an ever-increasing range to choose values from so that you avoid clustered client requests a.k.a. the thundering herd.